File: src/types/SyncResponse.ts
import FeedType from './FeedType';
import ImageStructure from './ImageStructure';

export interface ArticleResponse {
  id: string;
  title: string;
  content: string;
  date: string;
  read: boolean;
  image: ImageStructure | null;
}

export default interface SyncResponse {
  identifier: string;
  type: Exclude<FeedType, null>;
  articles: ArticleResponse[];
  image: ImageStructure | null;
}

File: src-tauri/src/structs/sync_response.rs
use serde::Serialize;
use crate::enums::feed_type::FeedType;
use crate::structs::article::Article;
use crate::structs::image::Image;

#[derive(Debug, Serialize)]
pub struct SyncResponse {
    pub identifier: String,
    #[serde(rename = "type")]
    pub feed_type: FeedType,
    pub articles: Vec<Article>,
    pub image: Option<Image>
}
